src/Budget.cpp

```diff
@@ -1,3 +1,70 @@
+//created By Ahmed Hamdy Kotp
 
-#include "../include/Budget.h"
+
+
+#include "Budget.h"
 #include <iostream>
+#include <iomanip>
+using namespace std;
+
+// Constructor 
+Budget::Budget() : total_budget(0.0) {}
+
+// Add a new budget category or update an existing one
+void Budget::add_category(const string& category_name) {
+    categories[category_name] = 0.0;
+}
+
+// Get the balance of a specific category
+double Budget::getBalance(const string& category_name) const {
+    auto bal = categories.find(category_name);
+    return bal != categories.end() ? bal->second : 0.0;
```

Comment:
if a category is not found, don't return zero as it's misleading (some existent categories may have 0 balance). Return any value that demonstrate that this category is not found (for example, -1);